### PR TITLE
fix(sdk): trigger save handler on question create when using SaveQuestionForm

### LINKED ISSUE #50008: onSave and onBeforeSave does not get called after a question is created in SaveQuestionForm sdk component
The `onSave` and `onBeforeSave` props on `<InteractiveQuestion />` never gets called even after the save button shows Success. This is because it internally calls `onCreate` and we don't call those handlers there, and we don't expose `onCreate`. We should make sure both `onSave` and `onBeforeSave` are called on question created.

We should also create an end-to-end test for this scenario.

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/50008

### Description

When using SaveQuestionForm to save question as a new question, the `onSave` and `onBeforeSave` handlers does not get called. This PR calls both handlers when a question is saved, and passes `{ isNewQuestion: true }` to the second argument of the save handler.

### How to verify

- Use the SaveQuestionForm to save question as a new question.
  - Refer to either [this PR's Storybook setup](https://github.com/metabase/metabase/pull/50137/files#diff-f8bc78b6ef233baa9bde9e81e3a24035d87e587a5ac333e3e2da6547c38986ceR31-R58) or https://github.com/metabase/shoppy/pull/86 (Shoppy PR)
- The `onSave` and `onBeforeSave` handlers must be called with `isNewQuestion: true`.

### Checklist

- [x] Tests have been added/updated to cover changes in this PR - E2E

### CODE REVIEW

**Review by deniskaber — APPROVED** [review]
Works good, but I think we should provide more value to onBeforeSave handler

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx [hunk 1]
```diff
@@ -64,16 +64,24 @@ export const InteractiveQuestionProvider = ({
...
     }
   };
 
   const handleCreate = async (question: Question) => {
-    await handleCreateQuestion(question);
-    await loadQuestion();
+    if (isSaveEnabled) {
+      const saveContext = { isNewQuestion: true };
+
+      await onBeforeSave?.(question, saveContext);
```
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx:80]: 💯, My best guess is @oisincoveney intended for it to show a loading indicator that a question is being saved. I like the idea of custom validation, but let's make sure the validation failed is reflected in the save form's UI. I think we already have a failed state in the save form, so we just have to trigger that.

My only concern is the `question` object is from `metabase-lib` and it contains MBQL stuff we might not want to expose, so I would rather add in validation once we have a public-facing question object ([something like this](https://www.notion.so/metabase/Tech-Allow-SDK-to-access-interactive-question-s-context-13e69354c9018098957df1a9dcc916f6?pvs=4#13e69354c901805fa5bcd3934e6dd8ce)).

[the remark this answers is not in the document]
Comment (oisincoveney) [reply enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx:80]: yeah it was something like that - let's give it enough info to make it work well 😄

**Review by oisincoveney — APPROVED** [review]

File: enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/context/InteractiveQuestionProvider.tsx:80]: task created with details: https://github.com/metabase/metabase/issues/50202

**Pull request merged**